packages/dataclass-editor/src/detail/field/browser/useTypeBrowser.tsx

```diff
@@ -2,22 +2,44 @@ import { BasicCheckbox, useBrowser, type Browser, type BrowserNode } from '@axon
 import { IvyIcons } from '@axonivy/ui-icons';
 import type { DataclassType } from '../../../protocol/types';
 import { typeBrowserApply } from './typeBrowserApply';
-import { useMemo, useState } from 'react';
+import { useEffect, useMemo, useState } from 'react';
 import { useMeta } from '../../../context/useMeta';
 import { typeData } from '../../../data/type-data';
 import { useAppContext } from '../../../context/AppContext';
 
 export const useTypeBrowser = (): Browser => {
   const { context } = useAppContext();
+  const [allTypesSearchActive, setAllTypesSearchActive] = useState<boolean>(false);
   const [typeAsList, setTypeAsList] = useState<boolean>(false);
+
   const dataClasses = useMeta('meta/scripting/dataClasses', context, []).data;
   const ivyTypes = useMeta('meta/scripting/ivyTypes', undefined, []).data;
-  const types = useMemo(() => typeData(dataClasses, ivyTypes), [dataClasses, ivyTypes]);
+
+  const [allTypesFilter, setAllTypesFilter] = useState('');
+  const ownTypes = useMeta('meta/scripting/ownTypes', { context, limit: 100, type: allTypesFilter }, []).data;
+  const { data: allDatatypes } = useMeta('meta/scripting/allTypes', { context, limit: 150, type: allTypesFilter }, []);
```

Comment:
Yes, that's correct. Now allTypes and ownTypes are always triggered with the new filter, regardless of the checkbox state. Disabling this might make sense, though I'm not sure if it is really that heavy, since we have limits on both. 
But perhaps @ivy-rew can provide more insight into the heaviness of this meta.

Maybe I could also just call setAllTypesFilter in the useEffect only when the checkbox is checked, instead of interducing a disabled-flag, right?

Reply:
I don't know why disable a query is bad or why use a useEffect should be better... Did you try it?